lib/runner.js

```diff
@@ -657,6 +661,30 @@ Runner.prototype.abort = function(){
 };
 
 /**
+ * Filter suites based on `isOnly` logic.
+ *
+ * @param {Array} suite
+ * @returns {Array}
+ * @api private
+ */
+function filterOnly(suite) {
+  // If it has `only` tests
+  if(suite.hasOnly) suite.tests = suite.hasOnly;
+  // If it doesn't has a suites make it running
+  // only if it mark as an `only`
+  if(!suite.suites.length) return suite.isOnly;
```

Comment:
It's used also as an iterator(recursively).

But, you right, it's a bug. I'll fix it quickly.
thx